### DISCUSSION TITLE
How to achieve user isolation in a chatflow?

### DISCUSSION THREAD

**Discussion by ops-004** [FlowiseAI/Flowise#1205] [Q&A]
Hello! I was impressed when I first used this tool. I hope to use it to build a chatflow and provide it for internal use within my company. However, I encountered an issue: I need to isolate the state of each user, such as managing their session history in the memory component. It is crucial that each user's session history remains isolated, as multiple users using it simultaneously may affect each other. Could you please advise me on how to solve this issue in such a scenario?

**Comment by HenryHengZJ** [comment]
You can use a memory node that has a `sessionId`. By using API you can then dynamically fill in the `sessionId` with anything. In your case, assuming each user has an id, use that as `sessionId`, this will isolate conversations between users
docs - https://docs.flowiseai.com/memory/zep-memory#use-in-flowise-api

**Reply by ops-004** [reply]
Thank you for your response. I didn't realize that OverrideConfig was so powerful!!!
